processing/src/main/java/io/druid/segment/ColumnSelectorFactory.java

```diff
@@ -27,6 +27,7 @@
 public interface ColumnSelectorFactory
 {
   public DimensionSelector makeDimensionSelector(DimensionSpec dimensionSpec);
+  public DimensionSelector makeDictEncodedStringDimensionSelector(DimensionSpec dimensionSpec);
```

Comment:
- I didn't see a compelling reason to make a new interface, I thought that DimensionSelector was sufficient as a "row reading" interface; if I made a new interface, it would look essentially the same in my mind. Creating a new interface seemed like it would result in unnecessary code churn given how many things use DimensionSelector.

- The handler could create its own DimensionSelector, it needs at a minimum, access to `EntryHolder currEntry`for IncrementalIndex and access to the current `cursorOffset` for QueryableIndex. 

I think this could be implemented as two makeDimensionSelector methods, on DimensionIndexer(for access to the dictionary within IncrementalIndex) and DimensionColumnReader (for access to column/dictionary within QueryableIndex).

I think I'll move the "makeDictEncodedStringDimensionSelector" functionality inside the String-specific dim interfaces and remove the direct dependency on ColumnSelectorFactory